### PR TITLE
fix: Database Connection Fix

### PR DESCRIPTION
**Pull request by PrudentBird**
added api controller
modified data source for cloud based postgres
added ssl config to data source
ssl defaults to false
cleaned up health controller
fixed lint errors

### CODE REVIEW

**Comment by PrudentBird** [comment]
@buka4rill kindly merge this PR

File: src/main.ts [hunk 1]
```diff
@@ -9,42 +9,50 @@ import dataSource, { initializeDataSource } from './database/data-source';
...
+    await seedingService.seedDatabase();
 
-  app.enable('trust proxy');
-  app.useLogger(logger);
-  app.enableCors();
-  app.setGlobalPrefix('api/v1', { exclude: ["/", "health"] });
+    app.enable('trust proxy');
+    app.useLogger(logger);
+    app.enableCors();
+    app.setGlobalPrefix('api/v1', { exclude: ['/', 'health', 'api', 'api/v1'] });
```
Comment (Homoakin619) [inline src/main.ts:33]: include "api/docs" in the exclude array

File: src/database/data-source.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ const dataSource = new DataSource({
   migrations: [process.env.DB_MIGRATIONS],
   synchronize: isDevelopment,
   migrationsTableName: 'migrations',
+  ssl: process.env.DB_SSL === 'true',
```
Comment (Homoakin619) [inline src/database/data-source.ts:19]: include this variable in the env.example file

**Comment by PrudentBird** [comment]
@Homoakin619 resolved !

**Comment by buka4rill** [comment]
Please let your PR name be descriptive of what it's actually doing

File: src/main.ts [hunk 3]
```diff
@@ -10,43 +10,50 @@ import { SeedingService } from './database/seeding/seeding.service';
...
+  } catch (err) {
+    console.error('Error during bootstrap', err);
+    process.exit(1);
+  }
 }
-bootstrap();
+bootstrap().catch(err => {
+  console.error('Error during bootstrap', err);
+  process.exit(1);
+});
```
Comment (buka4rill) [inline src/main.ts:59]: This change wasn't necessary

File: src/main.ts [hunk 4]
```diff
@@ -10,43 +10,50 @@ import { SeedingService } from './database/seeding/seeding.service';
 import { ResponseInterceptor } from './shared/inteceptors/response.interceptor';
```
Comment (buka4rill) [inline src/main.ts:10]: Why are your changes adding space to the linting?

**Review by buka4rill — COMMENTED** [review]
What's up with your linting?

**Comment by PrudentBird** [comment]
@buka4rill 

The linting is fixed now

File: src/main.ts (hunk 3 above)
Comment (buka4rill) [reply src/main.ts:59]: How so?

File: src/main.ts (hunk 4 above)
Comment (buka4rill) [reply src/main.ts:10]: I still don't see this fixed

File: src/main.ts (hunk 3 above)
Comment (PrudentBird) [reply src/main.ts:59]: image

**Comment by Homoakin619** [comment]
Your PR title here looks absurd, use a descriptive title

**Comment by PrudentBird** [comment]
@buka4rill kindly check again

**Review by Homoakin619 — APPROVED** [review]

File: src/api.controller.ts
Comment (Homoakin619) [inline src/api.controller.ts:1]: move these endpoints into the HealthController and delete this file

**Review by buka4rill — APPROVED** [review]

**Pull request merged**